src/datanode/src/instance/sql.rs

```diff
@@ -33,12 +33,23 @@ use crate::metric;
 use crate::sql::SqlRequest;
 
 impl Instance {
-    pub async fn execute_sql(&self, sql: &str, query_ctx: QueryContextRef) -> Result<Output> {
-        let stmt = self
+    pub async fn execute_sql(&self, sql: &str, query_ctx: QueryContextRef) -> Result<Vec<Output>> {
+        let stmts = self
             .query_engine
             .sql_to_statement(sql)
             .context(ExecuteSqlSnafu)?;
 
+        let mut results = Vec::with_capacity(stmts.len());
+
+        for stmt in stmts.into_iter() {
```

Comment:
Looks like `into_iter()` is redundant
```suggestion
        for stmt in stmts {
```

Reply:
good catch.